### PR TITLE
Read exclude patterns from .gitignore in absence of user-provided patterns (#344)

### PR DESCRIPTION
**Pull request by whosayn**
## Description

Use .gitignore to exclude files if --exclude is missing from both
pyproject.toml and the command line. Vulture now requires the
pathspec library to run.

## Related Issue

#344 

## Checklist:

- [x] I have updated the documentation in the README.md file or my changes don't require an update.
- [x] I have added an entry in CHANGELOG.md.
- [x] I have added or adapted tests to cover my changes.
- [x] I have run `tox -e fix-style` to format my code and checked the result with `tox -e style`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +83,10 @@ If you want to ignore a whole file or directory, use the `--exclude` parameter
 (e.g., `--exclude "*settings.py,*/docs/*.py,*/test_*.py,*/.venv/*.py"`). The
 exclude patterns are matched against absolute paths.
 
+Note that releases after 2.10 will parse the project's .gitignore files for
```
Comment (jendrikseipp) [inline README.md:86]:
```suggestion
Vulture 2.11+ parses the `.gitignore` file in the current working directory for
```

**Review by jendrikseipp — CHANGES_REQUESTED** [review]
Code looks great! After taking care of my nitpicks and fixing the CI checks, this is ready to be merged.

**Comment by whosayn** [comment]
@jendrikseipp a little stumped with the CI failures as I can't reproduce it locally. any ideas?

**Comment by jendrikseipp** [comment]
Thanks for making the revisions! I fixed the CI error and everything looks good now :-)

**Pull request merged**